File: binding/src/svg/libqsvgrenderer.cpp
#include <QByteArray>
#include <QObject>
#include <QPainter>
#include <QRect>
#include <QRectF>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QSvgRenderer>
#include <QTransform>
#include <QXmlStreamReader>
#include <qsvgrenderer.h>
#include "libqsvgrenderer.h"
#include "libqsvgrenderer.hxx"

QSvgRenderer* QSvgRenderer_new() {
	 return new QSvgRenderer();
}

QSvgRenderer* QSvgRenderer_new2(const libqt_string filename) {
	 return new QSvgRenderer(QString::fromUtf8(filename.data, filename.len));
}

QSvgRenderer* QSvgRenderer_new3(const libqt_string contents) {
	 return new QSvgRenderer(QByteArray(contents.data, contents.len));
}

QSvgRenderer* QSvgRenderer_new4(QXmlStreamReader* contents) {
	 return new QSvgRenderer(contents);
}

QSvgRenderer* QSvgRenderer_new5(QObject* parent) {
	 return new QSvgRenderer(parent);
}

QSvgRenderer* QSvgRenderer_new6(const libqt_string filename, QObject* parent) {
	 return new QSvgRenderer(QString::fromUtf8(filename.data, filename.len), parent);
}

QSvgRenderer* QSvgRenderer_new7(const libqt_string contents, QObject* parent) {
	 return new QSvgRenderer(QByteArray(contents.data, contents.len), parent);
}

QSvgRenderer* QSvgRenderer_new8(QXmlStreamReader* contents, QObject* parent) {
	 return new QSvgRenderer(contents, parent);
}

libqt_string QSvgRenderer_Tr(const char* s) {
	QString _ret = QSvgRenderer::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QSvgRenderer_IsValid(const QSvgRenderer* self) {
	return self->isValid();
}

QSize* QSvgRenderer_DefaultSize(const QSvgRenderer* self) {
	return new QSize(self->defaultSize());
}

QRect* QSvgRenderer_ViewBox(const QSvgRenderer* self) {
	return new QRect(self->viewBox());
}

QRectF* QSvgRenderer_ViewBoxF(const QSvgRenderer* self) {
	return new QRectF(self->viewBoxF());
}

void QSvgRenderer_SetViewBox(QSvgRenderer* self, const QRect* viewbox) {
	self->setViewBox(*viewbox);
}

void QSvgRenderer_SetViewBox2(QSvgRenderer* self, const QRectF* viewbox) {
	self->setViewBox(*viewbox);
}

int QSvgRenderer_AspectRatioMode(const QSvgRenderer* self) {
	return self->aspectRatioMode();
}

void QSvgRenderer_SetAspectRatioMode(QSvgRenderer* self, int mode) {
	self->setAspectRatioMode(static_cast<Qt::AspectRatioMode>(mode));
}

quint32 QSvgRenderer_Options(const QSvgRenderer* self) {
	return self->options();
}

void QSvgRenderer_SetOptions(QSvgRenderer* self, quint32 flags) {
	self->setOptions(static_cast<QFlags<QtSvg::Option>>(flags));
}

bool QSvgRenderer_Animated(const QSvgRenderer* self) {
	return self->animated();
}

int QSvgRenderer_FramesPerSecond(const QSvgRenderer* self) {
	return self->framesPerSecond();
}

void QSvgRenderer_SetFramesPerSecond(QSvgRenderer* self, int num) {
	self->setFramesPerSecond(num);
}

int QSvgRenderer_CurrentFrame(const QSvgRenderer* self) {
	return self->currentFrame();
}

void QSvgRenderer_SetCurrentFrame(QSvgRenderer* self, int currentFrame) {
	self->setCurrentFrame(currentFrame);
}

int QSvgRenderer_AnimationDuration(const QSvgRenderer* self) {
	return self->animationDuration();
}

bool QSvgRenderer_IsAnimationEnabled(const QSvgRenderer* self) {
	return self->isAnimationEnabled();
}

void QSvgRenderer_SetAnimationEnabled(QSvgRenderer* self, bool enable) {
	self->setAnimationEnabled(enable);
}

QRectF* QSvgRenderer_BoundsOnElement(const QSvgRenderer* self, const libqt_string id) {
	return new QRectF(self->boundsOnElement(QString::fromUtf8(id.data, id.len)));
}

bool QSvgRenderer_ElementExists(const QSvgRenderer* self, const libqt_string id) {
	return self->elementExists(QString::fromUtf8(id.data, id.len));
}

QTransform* QSvgRenderer_TransformForElement(const QSvgRenderer* self, const libqt_string id) {
	return new QTransform(self->transformForElement(QString::fromUtf8(id.data, id.len)));
}

void QSvgRenderer_SetDefaultOptions(quint32 flags) {
	QSvgRenderer::setDefaultOptions(static_cast<QFlags<QtSvg::Option>>(flags));
}

bool QSvgRenderer_Load(QSvgRenderer* self, const libqt_string filename) {
	return self->load(QString::fromUtf8(filename.data, filename.len));
}

bool QSvgRenderer_Load2(QSvgRenderer* self, const libqt_string contents) {
	return self->load(QByteArray(contents.data, contents.len));
}

bool QSvgRenderer_Load3(QSvgRenderer* self, QXmlStreamReader* contents) {
	return self->load(contents);
}

void QSvgRenderer_Render(QSvgRenderer* self, QPainter* p) {
	self->render(p);
}

void QSvgRenderer_Render2(QSvgRenderer* self, QPainter* p, const QRectF* bounds) {
	self->render(p, *bounds);
}

void QSvgRenderer_Render3(QSvgRenderer* self, QPainter* p, const libqt_string elementId) {
	self->render(p, QString::fromUtf8(elementId.data, elementId.len));
}

void QSvgRenderer_RepaintNeeded(QSvgRenderer* self) {
	self->repaintNeeded();
}

void QSvgRenderer_Connect_RepaintNeeded(QSvgRenderer* self, intptr_t slot) {
    void (*slotFunc)(QSvgRenderer*) = reinterpret_cast<void (*)(QSvgRenderer*)>(slot);
    QSvgRenderer::connect(self, &QSvgRenderer::repaintNeeded, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QSvgRenderer_Tr2(const char* s, const char* c) {
	QString _ret = QSvgRenderer::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QSvgRenderer_Tr3(const char* s, const char* c, int n) {
	QString _ret = QSvgRenderer::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QSvgRenderer_Render32(QSvgRenderer* self, QPainter* p, const libqt_string elementId, const QRectF* bounds) {
	self->render(p, QString::fromUtf8(elementId.data, elementId.len), *bounds);
}

void QSvgRenderer_Delete(QSvgRenderer* self) {
    delete self;
}

File: binding/include/svg/libqsvgrenderer.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qsvgrenderer.html
pub const qsvgrenderer = struct {

    /// New constructs a new QSvgRenderer object.
    pub fn New() QtC.QSvgRenderer {
        return qtc.QSvgRenderer_new();
    }


    /// New2 constructs a new QSvgRenderer object.
    pub fn New2(filename: []const u8) QtC.QSvgRenderer {
        const filename_str = qtc.libqt_string{
    .len = filename.len,
    .data = filename.ptr,
};
return qtc.QSvgRenderer_new2(filename_str);
    }


    /// New3 constructs a new QSvgRenderer object.
    pub fn New3(contents: []u8) QtC.QSvgRenderer {
        const contents_str = qtc.libqt_string{
    .len = contents.len,
    .data = contents.ptr,
};
return qtc.QSvgRenderer_new3(contents_str);
    }


    /// New4 constructs a new QSvgRenderer object.
    pub fn New4(contents: ?*anyopaque) QtC.QSvgRenderer {
        return qtc.QSvgRenderer_new4(@ptrCast(contents));
    }


    /// New5 constructs a new QSvgRenderer object.
    pub fn New5(parent: ?*anyopaque) QtC.QSvgRenderer {
        return qtc.QSvgRenderer_new5(@ptrCast(parent));
    }


    /// New6 constructs a new QSvgRenderer object.
    pub fn New6(filename: []const u8, parent: ?*anyopaque) QtC.QSvgRenderer {
        const filename_str = qtc.libqt_string{
    .len = filename.len,
    .data = filename.ptr,
};
return qtc.QSvgRenderer_new6(filename_str, @ptrCast(parent));
    }


    /// New7 constructs a new QSvgRenderer object.
    pub fn New7(contents: []u8, parent: ?*anyopaque) QtC.QSvgRenderer {
        const contents_str = qtc.libqt_string{
    .len = contents.len,
    .data = contents.ptr,
};
return qtc.QSvgRenderer_new7(contents_str, @ptrCast(parent));
    }


    /// New8 constructs a new QSvgRenderer object.
    pub fn New8(contents: ?*anyopaque, parent: ?*anyopaque) QtC.QSvgRenderer {
        return qtc.QSvgRenderer_new8(@ptrCast(contents), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QSvgRenderer_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qsvgrenderer.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QSvgRenderer_IsValid(@ptrCast(self));
    }

    pub fn DefaultSize(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QSvgRenderer_DefaultSize(@ptrCast(self));
    }

    pub fn ViewBox(self: ?*anyopaque, ) QtC.QRect {
        return qtc.QSvgRenderer_ViewBox(@ptrCast(self));
    }

    pub fn ViewBoxF(self: ?*anyopaque, ) QtC.QRectF {
        return qtc.QSvgRenderer_ViewBoxF(@ptrCast(self));
    }

    pub fn SetViewBox(self: ?*anyopaque, viewbox: ?*anyopaque) void {
        qtc.QSvgRenderer_SetViewBox(@ptrCast(self), @ptrCast(viewbox));
    }

    pub fn SetViewBox2(self: ?*anyopaque, viewbox: ?*anyopaque) void {
        qtc.QSvgRenderer_SetViewBox2(@ptrCast(self), @ptrCast(viewbox));
    }

    pub fn AspectRatioMode(self: ?*anyopaque, ) i32 {
        return qtc.QSvgRenderer_AspectRatioMode(@ptrCast(self));
    }

    pub fn SetAspectRatioMode(self: ?*anyopaque, mode: i32) void {
        qtc.QSvgRenderer_SetAspectRatioMode(@ptrCast(self), @intCast(mode));
    }

    pub fn Options(self: ?*anyopaque, ) u32 {
        return qtc.QSvgRenderer_Options(@ptrCast(self));
    }

    pub fn SetOptions(self: ?*anyopaque, flags: u32) void {
        qtc.QSvgRenderer_SetOptions(@ptrCast(self), @intCast(flags));
    }

    pub fn Animated(self: ?*anyopaque, ) bool {
        return qtc.QSvgRenderer_Animated(@ptrCast(self));
    }

    pub fn FramesPerSecond(self: ?*anyopaque, ) i32 {
        return qtc.QSvgRenderer_FramesPerSecond(@ptrCast(self));
    }

    pub fn SetFramesPerSecond(self: ?*anyopaque, num: i32) void {
        qtc.QSvgRenderer_SetFramesPerSecond(@ptrCast(self), num);
    }

    pub fn CurrentFrame(self: ?*anyopaque, ) i32 {
        return qtc.QSvgRenderer_CurrentFrame(@ptrCast(self));
    }

    pub fn SetCurrentFrame(self: ?*anyopaque, currentFrame: i32) void {
        qtc.QSvgRenderer_SetCurrentFrame(@ptrCast(self), currentFrame);
    }

    pub fn AnimationDuration(self: ?*anyopaque, ) i32 {
        return qtc.QSvgRenderer_AnimationDuration(@ptrCast(self));
    }

    pub fn IsAnimationEnabled(self: ?*anyopaque, ) bool {
        return qtc.QSvgRenderer_IsAnimationEnabled(@ptrCast(self));
    }

    pub fn SetAnimationEnabled(self: ?*anyopaque, enable: bool) void {
        qtc.QSvgRenderer_SetAnimationEnabled(@ptrCast(self), enable);
    }

    pub fn BoundsOnElement(self: ?*anyopaque, id: []const u8) QtC.QRectF {
        const id_str = qtc.libqt_string{
    .len = id.len,
    .data = id.ptr,
};
return qtc.QSvgRenderer_BoundsOnElement(@ptrCast(self), id_str);
    }

    pub fn ElementExists(self: ?*anyopaque, id: []const u8) bool {
        const id_str = qtc.libqt_string{
    .len = id.len,
    .data = id.ptr,
};
return qtc.QSvgRenderer_ElementExists(@ptrCast(self), id_str);
    }

    pub fn TransformForElement(self: ?*anyopaque, id: []const u8) QtC.QTransform {
        const id_str = qtc.libqt_string{
    .len = id.len,
    .data = id.ptr,
};
return qtc.QSvgRenderer_TransformForElement(@ptrCast(self), id_str);
    }

    pub fn SetDefaultOptions(flags: u32) void {
        qtc.QSvgRenderer_SetDefaultOptions(@intCast(flags));
    }

    pub fn Load(self: ?*anyopaque, filename: []const u8) bool {
        const filename_str = qtc.libqt_string{
    .len = filename.len,
    .data = filename.ptr,
};
return qtc.QSvgRenderer_Load(@ptrCast(self), filename_str);
    }

    pub fn Load2(self: ?*anyopaque, contents: []u8) bool {
        const contents_str = qtc.libqt_string{
    .len = contents.len,
    .data = contents.ptr,
};
return qtc.QSvgRenderer_Load2(@ptrCast(self), contents_str);
    }

    pub fn Load3(self: ?*anyopaque, contents: ?*anyopaque) bool {
        return qtc.QSvgRenderer_Load3(@ptrCast(self), @ptrCast(contents));
    }

    pub fn Render(self: ?*anyopaque, p: ?*anyopaque) void {
        qtc.QSvgRenderer_Render(@ptrCast(self), @ptrCast(p));
    }

    pub fn Render2(self: ?*anyopaque, p: ?*anyopaque, bounds: ?*anyopaque) void {
        qtc.QSvgRenderer_Render2(@ptrCast(self), @ptrCast(p), @ptrCast(bounds));
    }

    pub fn Render3(self: ?*anyopaque, p: ?*anyopaque, elementId: []const u8) void {
        const elementId_str = qtc.libqt_string{
    .len = elementId.len,
    .data = elementId.ptr,
};
qtc.QSvgRenderer_Render3(@ptrCast(self), @ptrCast(p), elementId_str);
    }

    pub fn RepaintNeeded(self: ?*anyopaque, ) void {
        qtc.QSvgRenderer_RepaintNeeded(@ptrCast(self));
    }

    pub fn OnRepaintNeeded(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QSvgRenderer_Connect_RepaintNeeded(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QSvgRenderer_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qsvgrenderer.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QSvgRenderer_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qsvgrenderer.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Render32(self: ?*anyopaque, p: ?*anyopaque, elementId: []const u8, bounds: ?*anyopaque) void {
        const elementId_str = qtc.libqt_string{
    .len = elementId.len,
    .data = elementId.ptr,
};
qtc.QSvgRenderer_Render32(@ptrCast(self), @ptrCast(p), elementId_str, @ptrCast(bounds));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSvgRenderer_Delete(@ptrCast(self));
    }
};
